include/nbl/builtin/hlsl/spirv_intrinsics/core.hlsl

```diff
@@ -47,199 +69,1272 @@ static const uint32_t3 LocalInvocationId;
+	}
+
+	void roundingModeRTPINTEL()
+	{
+		vk::ext_execution_mode(spv::ExecutionModeRoundingModeRTPINTEL);
+	}
+
+	void roundingModeRTNINTEL()
+	{
+		vk::ext_execution_mode(spv::ExecutionModeRoundingModeRTNINTEL);
+	}
+
+	void floatingPointModeALTINTEL()
+	{
+		vk::ext_execution_mode(spv::ExecutionModeFloatingPointModeALTINTEL);
+	}
+
+	void floatingPointModeIEEEINTEL()
+	{
+		vk::ext_execution_mode(spv::ExecutionModeFloatingPointModeIEEEINTEL);
+	}
+
+	void maxWorkgroupSizeINTEL()
+	{
+		vk::ext_execution_mode(spv::ExecutionModeMaxWorkgroupSizeINTEL);
+	}
+
+	void maxWorkDimINTEL()
+	{
+		vk::ext_execution_mode(spv::ExecutionModeMaxWorkDimINTEL);
+	}
+
+	void noGlobalOffsetINTEL()
+	{
+		vk::ext_execution_mode(spv::ExecutionModeNoGlobalOffsetINTEL);
+	}
+
+	void numSIMDWorkitemsINTEL()
+	{
+		vk::ext_execution_mode(spv::ExecutionModeNumSIMDWorkitemsINTEL);
+	}
+
+	void schedulerTargetFmaxMhzINTEL()
+	{
+		vk::ext_execution_mode(spv::ExecutionModeSchedulerTargetFmaxMhzINTEL);
+	}
+
+	void maximallyReconvergesKHR()
+	{
+		vk::ext_execution_mode(spv::ExecutionModeMaximallyReconvergesKHR);
+	}
+
+	void fPFastMathDefault()
+	{
+		vk::ext_execution_mode(spv::ExecutionModeFPFastMathDefault);
+	}
+
+	void streamingInterfaceINTEL()
+	{
+		vk::ext_execution_mode(spv::ExecutionModeStreamingInterfaceINTEL);
+	}
+
+	void registerMapInterfaceINTEL()
+	{
+		vk::ext_execution_mode(spv::ExecutionModeRegisterMapInterfaceINTEL);
+	}
+
+	void namedBarrierCountINTEL()
+	{
+		vk::ext_execution_mode(spv::ExecutionModeNamedBarrierCountINTEL);
+	}
+
+	void maximumRegistersINTEL()
+	{
+		vk::ext_execution_mode(spv::ExecutionModeMaximumRegistersINTEL);
+	}
+
+	void maximumRegistersIdINTEL()
+	{
+		vk::ext_execution_mode(spv::ExecutionModeMaximumRegistersIdINTEL);
+	}
+
+	void namedMaximumRegistersINTEL()
+	{
+		vk::ext_execution_mode(spv::ExecutionModeNamedMaximumRegistersINTEL);
+	}
+}
+
+//! Group Operations
+namespace group_operation
+{
+	static const uint32_t Reduce = 0;
+	static const uint32_t InclusiveScan = 1;
+	static const uint32_t ExclusiveScan = 2;
+	static const uint32_t ClusteredReduce = 3;
+	static const uint32_t PartitionedReduceNV = 6;
+	static const uint32_t PartitionedInclusiveScanNV = 7;
+	static const uint32_t PartitionedExclusiveScanNV = 8;
+}
+
+//! Instructions
+template<typename T, typename P>
+[[vk::ext_instruction(spv::OpLoad)]]
+enable_if_t<is_spirv_type_v<P>, T> load(P pointer, [[vk::ext_literal]] uint32_t memoryAccess);
+
+template<typename T, typename P>
+[[vk::ext_instruction(spv::OpLoad)]]
+enable_if_t<is_spirv_type_v<P>, T> load(P pointer, [[vk::ext_literal]] uint32_t memoryAccess, [[vk::ext_literal]] uint32_t memoryAccessParam);
+
+template<typename T, typename P, uint32_t alignment>
+[[vk::ext_instruction(spv::OpLoad)]]
+enable_if_t<is_spirv_type_v<P>, T> load(P pointer, [[vk::ext_literal]] uint32_t __aligned = /*Aligned*/0x00000002, [[vk::ext_literal]] uint32_t __alignment = alignment);
```

Comment:
why do you gen these overloads?

Reply:
it's defined in the grammer. [`MemoryAccess` operand](https://github.com/KhronosGroup/SPIRV-Headers/blob/efb6b4099ddb8fa60f62956dee592c4b94ec6a49/include/spirv/unified1/spirv.core.grammar.json#L719) may have [zero or more parameters](https://github.com/KhronosGroup/SPIRV-Headers/blob/efb6b4099ddb8fa60f62956dee592c4b94ec6a49/include/spirv/unified1/spirv.core.grammar.json#L10642-L10741).